code/modules/vehicles/segway.dm

```diff
@@ -0,0 +1,221 @@
+/obj/vehicle/segway
+	name = "security segway"
+	desc = "For the slow security or mall cops"
+	icon = 'icons/obj/segway.dmi'
+	icon_state = "segway"
+	dir = SOUTH
+
+	load_item_visible = 1
+	mob_offset_y = 5
+	health = 120
+	maxhealth = 120
+
+	fire_dam_coeff = 0.6
+	brute_dam_coeff = 0.5
+	var/protection_percent = 60
+
+	var/land_speed = 4 //if 0 it can't go on turf
+	var/space_speed = 8
+	var/segway_icon = "segway"
+
+	var/datum/effect_system/ion_trail/ion
+	var/kickstand = 1
+	var/obj/item/weapon/key/segwaykey/key
+
+/obj/item/weapon/key/segwaykey
+	name = "key"
+	desc = "A small key fob, a ISD symbol is engraved into it."
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "segwaykeys"
+	w_class = 1
+
+/obj/vehicle/segway/Initialize()
+	. = ..()
+	ion = new(src)
+	cell = new /obj/item/weapon/cell/high(src)
+	turn_off()
+
+/obj/vehicle/segway/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/key/segwaykey))
+		if(!key)
+			user.drop_from_inventory(W,src)
+			key = W
+			verbs += /obj/vehicle/train/cargo/engine/verb/remove_key
+		return
+	..()
+
+/obj/vehicle/segway/turn_on()
```

Comment:
This proc is doing something horribly wrong. We're adding start_engine, but we should already have it no...? And we're potentially adding start_engine twice?